subsys/logging/log_core.c

```diff
@@ -644,6 +647,12 @@ char *log_strdup(const char *str)
 		return (char *)log_strdup_fail_msg;
 	}
 
+	if (IS_ENABLED(CONFIG_LOG_STRDUP_POOL_PROFILING)) {
+		atomic_inc(&log_strdup_in_use);
+		log_strdup_max = MAX(log_strdup_in_use, log_strdup_max);
```

Comment:
Access to log_strdup_in_use is not atomic inside MAX().

Reply:
tried to fix that.

Reply:
I suppose `atomic_get()` should be used

Reply:
But then there will be 2 calls to atomic_get(), so the problem is still not solved.
```
val = atomic_get(..)
max = MAX(val, max)
```

Reply:
right

Reply:
i've ended up using `irq_lock`. I managed to handle `log_strdup_max` without locks but handling `log_strdup_longest` is more tricky. Maybe it could be done with multiple while loops but then i've measured how much time it takes and it was more than 3 times longer than with lock so that would be art for art only.